src/common/pick_address.cc

```diff
@@ -285,6 +285,33 @@ static int fill_in_one_address(
   return 0;
 }
 
+bool networks_cover_address_family(CephContext *cct, const std::string &networks, unsigned ipv) {
+  std::list<string> nets;
+  get_str_list(networks, nets);
+  bool has_ipv4 = !(ipv & CEPH_PICK_ADDRESS_IPV4);
+  bool has_ipv6 = !(ipv & CEPH_PICK_ADDRESS_IPV6);
+
+  for (auto& s : nets) {
+    struct sockaddr_storage net;
+    unsigned prefix_len;
+    if (!parse_network(s.c_str(), &net, &prefix_len)) {
+      lderr(cct) << "unable to parse network: " << s << dendl;
+      exit(1);
+    }
+
+    switch (net.ss_family) {
+    case AF_INET:
+      has_ipv4 = true;
+      break;
+    case AF_INET6:
+      has_ipv6 = true;
+      break;
+    }
+  }
```

Comment:
it's a clever way to check if the `nets` covers IPv4 and/or IPv6. but it's confusing and difficult to reason about.
```suggestion
    unsigned found_ipv = 0;
    switch (net.ss_family) {
    case AF_INET:
      found_ipv |= CEPH_PICK_ADDRESS_IPV4;
      break;
    case AF_INET6:
      found_ipv |= CEPH_PICK_ADDRESS_IPV6;
      break;
    }
  }
  return found_ipv;
```

Reply:
Oh that's a pretty neat way of dealing with it. Only query I have is as the function currently stands it's stand alone and returns yes or no you have the networks you need. In your approach, we get an int with the required bits set, but then we need to then go check the result outside of the method. So only does 1/2 the job.

I guess that's ok, as we now have an int that represents what networks we cover based on what was gave. I guess my brain works differently, maybe too much python :P I find my way easier to understand. But your way does fit with how ceph is managing with the ipv, so we should keep it consistent. I'll will go make the change :)